examples/all-clusters-app/linux/main-common.cpp

```diff
@@ -41,6 +44,200 @@ using namespace chip::DeviceLayer;
 
 namespace {
 static LowPowerManager lowPowerManager;
+
+bool EnabledWithServerCluster(ClusterId clusterId)
+{
+    bool retval = false;
+
+    for (auto endpointId : EnabledEndpointsWithServerCluster(clusterId))
+    {
+        IgnoreUnusedVariable(endpointId);
+        retval = true;
+    }
+
+    return retval;
+}
+
+/**
+ * Should be called when a software fault takes place on the Node.
+ */
+void HandleSoftwareFaultEvent(intptr_t arg)
+{
+    if (!EnabledWithServerCluster(Clusters::SoftwareDiagnostics::Id))
+        return;
+
+    Clusters::SoftwareDiagnostics::Structs::SoftwareFaultStruct::Type softwareFault;
+    char threadName[kMaxThreadNameLength + 1];
+
+    softwareFault.id = static_cast<uint64_t>(gettid());
+    Platform::CopyString(threadName, kMaxThreadNameLength + 1, std::to_string(softwareFault.id).c_str());
+
+    softwareFault.name           = CharSpan::fromCharString(threadName);
+    softwareFault.faultRecording = ByteSpan(Uint8::from_const_char(threadName), strlen(threadName));
```

Comment:
Again, the problem is that this going to make people think the faultRecording needs to match the name...